catalog/dags/providers/provider_api_scripts/wikimedia_commons.py

```diff
@@ -320,7 +320,7 @@ def get_record_data(self, record):
         creator, creator_url = self.extract_creator_info(media_info)
         title = self.extract_title(media_info)
         filesize = media_info.get("size", 0)  # in bytes
-        filetype = self.extract_file_type(url, valid_media_type)
+        filetype, _ = extract_filetype(url)
```

Comment:
I still don't really understand why we should throw away the media_type value from this and keep the `::extract_media_type` method. It would be a lot clearer if these calls were next to each other and an explanation in comment were added to clarify their relationship to one another. Even knowing the context of this PR, I would struggle to explain the reasoning behind continuing to use the Wikimedia media types.

Especially because `::extract_media_type` _already extracts the mimetype Wikimedia has for the file_, which should in theory be more reliable than extracting it from the URL extension.

I really strongly feel these should not be separate methods. Either we should just use the Wikimedia version (and extract the "filetype" we want from it) or only use `extract_filetype` and ignore the Wikimedia metadata (the latter of which feels very wrong to me, like we are ignoring the authority). At worst we should fall back to `extract_filetype` if for some reason the Wikimedia metadata doesn't make sense.

Two separate methods for this is very confusing and at minimum warrants a comprehensive explanation. But really I think we should consolidate these into a single routine. The code is really confusing as it is now (and already was before, FWIW!).

Reply:
> I still don't really understand why we should throw away the media_type value from this and keep the ::extract_media_type method. It would be a lot clearer if these calls were next to each other and an explanation in comment were added to clarify their relationship to one another. Even knowing the context of this PR, I would struggle to explain the reasoning behind continuing to use the Wikimedia media types.

I changed the assumption in that Wikimedia classification is correct after [Olga's comment](https://github.com/WordPress/openverse/pull/4785#issuecomment-2301909971), and then the media_type from `extract_filetype` theoretically will always be the same than the returned from the Wikimedia specific method, `::extract_media_type`.

> Especially because ::extract_media_type already extracts the mimetype Wikimedia has for the file, which should in theory be more reliable than extracting it from the URL extension.

That Wikimedia method only extracts the media type, whether the record is an image or audio, not so the mime type. But this gave me an idea! It would be a lot easier if we could get the mime type [directly from the API](https://commons.wikimedia.org/w/api.php?action=help&modules=query%2Bimageinfo), and actually we can! I think that would be more reliable. 

I made the changes for this locally, just need to modify tests. Wait for it 🤞

Reply:
I see, okay. I misunderstood the code in `extract_media_type`. I thought the check, `if media_type in image_mediatypes:`, was looking at the MediaWiki list of image mime types, but it's _media_ types. Sorry for the confusion I've caused.

I'm glad there is a way to get the mime type from Wikimedia, though. Hopefully it will be reliable and help consolidate this logic.